### PR TITLE
Samreen Zehra code review changes

### PR DESCRIPTION
**Pull request by SamreenZ13**
Summarize the three technical changes you made, and why.
1. Removed the unused usings from the project .cs classes.
Reason: This would ensure an increase in compile-time and better code readability as only relevant references are present.
2. Removed the Yelp. , Business. from various parts of the code and added to the usings.
Reason: It makes the code cleaner and more user-readable.
3. Combined the code for fetching yelp business into an array and then into a list. 
Reason: It can be done in a single line of code instead of first fetching it as an array and then separately casting it to a list collection object. 
4. In the model classes added variables as private members and used the properties to access the variables. 
Reason: This would ensure the member variables are not accessed/modified outside the class.
5. Added a constant file to store all the constant variables and created constants for few hardcoded values.
 Reason: this is a good coding practice to store all the constant and hardcoded values in one place so that when needed to be modified they have to be changed only in the constant file and not across. Ensures consistency as well.

List three items you learned from the code review.
1. How to use inheritance to implement a model class.
2. I learned how to get familiarized with a new piece of code and refactor it
3. I learned how to do a code review and via GitHub by creating a branch and a pull request(Since I am new to GitHub).

Was the project available in GitHub on time?
Yes 

Does the project compile?
Yes 

Does the project have adequate documentation?
Yes

### CODE REVIEW

File: BreweryRatings/BusinessReview.cs [hunk 1]
```diff
@@ -17,50 +17,66 @@ namespace ReviewNamespace
 
     public partial class BusinessReview
     {
+        private Review[] _reviews;
+        private long _total;
+        private string[] _possibleLanguage;
+
         [JsonProperty("reviews")]
-        public Review[] Reviews { get; set; }
+        public Review[] Reviews { get => _reviews; set => _reviews = value; }
```
Comment (discospiff) [inline BreweryRatings/BusinessReview.cs:25]: Encapsulation is a good idea, though {get; set; } syntax is often considered a shortcut for encapsulation.  In our class, I'm fine with either approach.

File: BreweryRatings/BusinessWithReview.cs [hunk 2]
```diff
@@ -14,17 +14,22 @@ namespace Yelp
     using System.Globalization;
     using Newtonsoft.Json;
     using Newtonsoft.Json.Converters;
+    using ReviewNamespace;
```
Comment (discospiff) [inline BreweryRatings/BusinessWithReview.cs:17]: Having the using in one place is much cleaner.  Good idea.

File: BreweryRatings/Pages/Index.cshtml.cs [hunk 3]
```diff
@@ -31,7 +29,7 @@ public void OnGet(string location)
         {
             if (string.IsNullOrEmpty(location))
             {
-                location = "Cincinnati";
+                location = Constant.DEFAULT_LOCATION;
```
Comment (discospiff) [inline BreweryRatings/Pages/Index.cshtml.cs:32]: Good... constants help to keep things clean.

**Comment by discospiff** [comment]
Several good ideas here... the team should review and consider merging.

**Comment by PauloCSoto** [comment]
That’s great, Sidney, thanks!

Today is my last day in Japan and I’ll be traveling back to the states on Sunday. I’ll be back home on Sunday night or Monday early afternoon so I can work on the project some more if need be.

Paulo

On Nov 29, 2019, at 10:19 PM, Sidney Schaeper <notifications@github.com> wrote:

﻿

Merged #3<https://github.com/PauloCSoto/BreweryRatings/pull/3> into master.

**Pull request merged**